### PR TITLE
Add tf_object_detection

### PR DESCRIPTION
**Pull request by adbeda**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [ ] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by adbeda** [comment]
@mbargull thanks for heads up [on the PR ](https://github.com/conda-forge/staged-recipes/pull/8590)that somehow got tangled up with some other PR. I created a new one off of `master` one to avoid any issue.

File: recipes/tf_object_detection/build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+cd research
+if [`uname` == Darwin];then
+    curl -OL https://github.com/google/protobuf/releases/download/v3.3.0/$PROTOC_ZIP
```
Comment (isuruf) [inline recipes/tf_object_detection/build.sh]: Use conda-build's multiple outputs to download these.

Comment (isuruf) [reply recipes/tf_object_detection/build.sh]: Also why not use `protobuf` package from conda-forge?

Comment (adbeda) [reply recipes/tf_object_detection/build.sh]: @isuruf I'm just converted the guide to[ installation.md](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/installation.md) to build.sh based on @jjhelmus recommendation

Comment (isuruf) [reply recipes/tf_object_detection/build.sh]: You can add `protobuf` as a dependency in host and only do `protoc object_detection/protos/*.proto --python_out=.` and then the install line.

Comment (adbeda) [reply recipes/tf_object_detection/build.sh]: got it

**Comment by adbeda** [comment]
@isuruf thanks for the heads up

**Pull request merged**